CNN.diff

```diff
@@ -0,0 +1,317 @@
+             self.reduce = False
+             
+-        elif args['model_size'] == 'small':
++        elif self.model_type == 'small':
+             d = {4: 135, 5: 511, 6: 2079}
+             self.n_features = d[args['k']]
+             self.net = myNet(self.n_features, args['n_clusters'])
+             self.reduce = True
+             
+-        elif args['model_size'] == 'full':
++        elif self.model_type == 'full':
+             self.net = ResNet18(1, args['n_clusters'])
++        elif self.model_type == 'conv':
++            self.net = ConvNet(1, args['n_clusters'])
++            self.reduce = False
+         else:
+             raise ValueError("Invalid Model Type")
+         
+@@ -101,24 +105,28 @@ class IID_model():
+         #Data Files
+         data_path = self.sequence_file
+         GT_file = self.GT_file
++        representation = 'fcgr' if self.model_type in ('conv') else 'vector'
+                 
+         self.dataloader = create_dataloader(data_path, 
+                                              self.n_mimics, 
+                                              k=self.k, 
+                                              batch_size=self.batch_sz, 
+                                              GT_file=GT_file,
+-                                             reduce=self.reduce)
++                                             reduce=self.reduce,
++                                             representation=representation)
+ 
+     def contrastive_training_epoch(self):
+-        n_features = self.n_features
+-        batch_size = self.batch_sz
+-        k = self.k 
+         self.net.train()
+         running_loss = 0.0
+ 
+         for i_batch, sample_batched in enumerate(self.dataloader):
+-            sample = sample_batched['true'].view(-1, 1, self.n_features).type(dtype)
+-            modified_sample = sample_batched['modified'].view(-1, 1, self.n_features).type(dtype)
++
++            if self.model_type not in ('conv'):
++                sample = sample_batched['true'].view(-1, 1, self.n_features).type(dtype)
++                modified_sample = sample_batched['modified'].view(-1, 1, self.n_features).type(dtype)
++            else:
++                sample = sample_batched['true'][:, np.newaxis, ...].type(dtype)
++                modified_sample = sample_batched['modified'][:, np.newaxis, ...].type(dtype)
+             
+             # zero the gradients
+             self.optimizer.zero_grad()
+@@ -145,8 +153,7 @@ class IID_model():
+ 
+     def predict(self, data=None):
+         
+-        n_features = self.n_features
+-        test_dataset = SequenceDataset(self.sequence_file, k=self.k, transform=None, GT_file=self.GT_file, reduce=self.reduce)
++        test_dataset = SequenceDataset(self.sequence_file, k=self.k, transform=None, GT_file=self.GT_file, reduce=self.reduce, representation=self.model_type)
+         test_dataloader = DataLoader(test_dataset, 
```

Comment:
representaion should not be either 'cgr' or 'vector' ?  why am I passing model type to representation in this class ?